dev/bots/test.dart

```diff
@@ -375,6 +382,7 @@ Future<void> _runExampleProjectBuildTests(FileSystemEntity exampleDirectory) asy
   final List<String> additionalArgs = hasNullSafety
     ? <String>['--enable-experiment', 'non-nullable', '--no-sound-null-safety']
     : <String>[];
+  final String branch = Platform.environment['CIRRUS_BRANCH'];
```

Comment:
I would move this into _shouldRunMacOS too since it doesn't look like its used elsewhere

Reply:
actually, does this need a different config for LUCI, or is the plan to only run these on cirrus?

Reply:
> I would move this into _shouldRunMacOS too since it doesn't look like its used elsewhere

Linux and Windows will have identical code here. I guess I could extract it to a branch getter though to share it that way though (which would also be useful for Cirrus/LUCI distinction)

> actually, does this need a different config for LUCI, or is the plan to only run these on cirrus?

This pattern was taken from `_runBuildTests()`, so I assumed something was setting it on LUCI for compatibility. Maybe that's not the case and those tests are still on Cirrus though?

@godofredoc Do you know what the right way to query the branch would be for LUCI?

Reply:
we added some functionality to pass LUCI_BRANCH to some builders, we can extend it to this builder. One caveat is that branch resolves to master/dev/beta/stable do you need the real branch here?

Reply:
This is just to ensure we're excluding desktop tests (and web tests; I guess those haven't migrated to LUCI yet since this script should already be broken otherwise?) on stable and beta, so if it's just one of those four strings that's fine.

Reply:
Could it be that the script is disabling those tests if the env variable CIRRUS_BRANCH doesn't exist? that may be the reason we never noticed that some tests were not running.

Reply:
The existing web logic is:
```
  final String branch = Platform.environment['CIRRUS_BRANCH'];	
  [...]
    if (branch != 'beta' && branch != 'stable')
```

A null/empty value would pass that check. 🤷 

Anyway, I'll update it and the new code to use either CIRRUS_BRANCH or LUCI_BRANCH depending on the builder.